File: packages/url_launcher/url_launcher/example/ios/RunnerUITests/URLLauncherUITests.m
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

@import XCTest;
@import os.log;

@interface URLLauncherUITests : XCTestCase
@property(nonatomic, strong) XCUIApplication* app;
@end

@implementation URLLauncherUITests

- (void)setUp {
  self.continueAfterFailure = NO;

  self.app = [[XCUIApplication alloc] init];
  [self.app launch];
}

- (void)testLaunch {
  XCUIApplication* app = self.app;

  NSArray<NSString*>* buttonNames = @[
    @"Launch in app", @"Launch in app(JavaScript ON)", @"Launch in app(DOM storage ON)",
    @"Launch a universal link in a native app, fallback to Safari.(Youtube)"
  ];
  for (NSString* buttonName in buttonNames) {
    XCUIElement* button = app.buttons[buttonName];
    if (![button waitForExistenceWithTimeout:30.0]) {
      os_log_error(OS_LOG_DEFAULT, "%@", app.debugDescription);
      XCTFail(@"Failed due to not able to find %@ button", buttonName);
    }
    XCTAssertEqual(app.webViews.count, 0);
    [button tap];
    XCUIElement* webView = app.webViews.firstMatch;
    if (![webView waitForExistenceWithTimeout:30.0]) {
      os_log_error(OS_LOG_DEFAULT, "%@", app.debugDescription);
      XCTFail(@"Failed due to not able to find webview");
    }
    XCTAssertTrue(app.buttons[@"ForwardButton"].exists);
    XCTAssertTrue(app.buttons[@"ShareButton"].exists);
    XCTAssertTrue(app.buttons[@"OpenInSafariButton"].exists);
    [app.buttons[@"Done"] tap];
  }
}

@end

File: packages/url_launcher/url_launcher_ios/example/ios/RunnerUITests/URLLauncherUITests.swift
//
//  URLLauncherUITests.swift
//  RunnerUITests
//
//  Created by Andres Gutierrez on 2/7/23.
//  Copyright © 2023 The Flutter Authors. All rights reserved.
//

import Foundation
import XCTest

class URLLauncherUITests: XCTestCase {
    override func setUp() {
        self.continueAfterFailure = false
        
        XCUIApplication().launch()
    }
    
    func testLaunch() {
        let app = XCUIApplication()
        
        let buttonNames = ["Launch in app", "Launch in app(JavaScript ON)", "Launch in app(DOM storage ON)", "Launch a universal link in a native app, fallback to Safari.(Youtube)"
        ]
        
        for buttonName in buttonNames {
            let button = app.buttons[buttonName]
            XCTAssertTrue(button.waitForExistence(timeout: 30.0))
            XCTAssertEqual(app.webViews.count, 0)
            
            button.tap()
            let webView = app.webViews.firstMatch
            XCTAssertTrue(webView.waitForExistence(timeout: 30.0))
            XCTAssertTrue(app.buttons["ForwardButton"].waitForExistence(timeout: 30.0))
            XCTAssertTrue(app.buttons["Share"].exists)
            XCTAssertTrue(app.buttons["OpenInSafariButton"].exists)
            
            app.buttons["Done"].tap()
        }
    }
}
